geochatt/__init__.py

```diff
@@ -76,20 +76,87 @@ def get_address(longitude, latitude, max_distance=0.0001):
+    "HEIGHTS": "HTS", "HIGHWAY": "HWY", "HILL": "HL", "HILLS": "HLS", "HOLLOW": "HOLW", "INLET": "INLT", 
+    "ISLAND": "IS", "ISLANDS": "ISS", "ISLE": "ISLE", "JUNCTION": "JCT", "JUNCTIONS": "JCTS", "KEY": "KY", 
+    "KEYS": "KYS", "KNOLL": "KNL", "LAKE": "LK", "LAKES": "LKS", "LAND": "LAND", "LANDING": "LNDG", 
+    "LANE": "LN", "LIGHT": "LGT", "LIGHTS": "LGTS", "LOAF": "LF", "LOCK": "LCK", "LOCKS": "LCKS", 
+    "LODGE": "LDG", "LOOP": "LOOP", "MALL": "MALL", "MANOR": "MNR", "MANORS": "MNRS", "MEADOW": "MDW", 
+    "MEADOWS": "MDWS", "MEWS": "MEWS", "MILL": "ML", "MILLS": "MLS", "MISSION": "MSN", "MOTORWAY": "MTWY", 
+    "MOUNT": "MT", "MOUNTAIN": "MTN", "MOUNTAINS": "MTNS", "NECK": "NCK", "ORCHARD": "ORCH", "OVAL": "OVAL", 
+    "OVERPASS": "OPAS", "PARK": "PARK", "PARKS": "PARK", "PARKWAY": "PKWY", "PARKWAYS": "PKWY", 
+    "PASS": "PASS", "PASSAGE": "PSGE", "PATH": "PATH", "PIKE": "PIKE", "PINE": "PNE", "PINES": "PNES", 
+    "PL": "PLACE", "PLAIN": "PLN", "PLAINS": "PLNS", "PLAZA": "PLZ", "POINT": "PT", "POINTS": "PTS", 
+    "PORT": "PRT", "PORTS": "PRTS", "PRAIRIE": "PR", "RADIAL": "RADL", "RAMP": "RAMP", "RANCH": "RNCH", 
+    "RAPID": "RPD", "RAPIDS": "RPDS", "REST": "RST", "RIDGE": "RDG", "RIDGES": "RDGS", "RIVER": "RIV", 
+    "ROAD": "RD", "ROADS": "RDS", "ROUTE": "RTE", "ROW": "ROW", "RUE": "RUE", "RUN": "RUN", "SHOAL": "SHL", 
+    "SHOALS": "SHLS", "SHORE": "SHR", "SHORES": "SHRS", "SKYWAY": "SKWY", "SPRING": "SPG", 
+    "SPRINGS": "SPGS", "SPUR": "SPUR", "SPURS": "SPUR", "SQUARE": "SQ", "SQUARES": "SQS", "STATION": "STA", 
+    "STRAVENUE": "STRA", "STREAM": "STRM", "STREET": "ST", "STREETS": "STS", "SUMMIT": "SMT", 
+    "TERRACE": "TER", "THROUGHWAY": "TRWY", "TRACE": "TRCE", "TRACK": "TRAK", "TRAFFICWAY": "TRFY", 
+    "TRAIL": "TRL", "TRAILER": "TRLR", "TUNNEL": "TUNL", "TURNPIKE": "TPKE", "UNDERPASS": "UPAS", 
+    "UNION": "UN", "UNIONS": "UNS", "VALLEY": "VLY", "VALLEYS": "VLYS", "VIADUCT": "VIA", "VIEW": "VW", 
+    "VIEWS": "VWS", "VILLAGE": "VLG", "VILLAGES": "VLGS", "VILLE": "VL", "VISTA": "VIS", "WALK": "WALK", 
+    "WALKS": "WALK", "WALL": "WALL", "WAY": "WAY", "WAYS": "WAYS", "WELL": "WL", "WELLS": "WLS"
+}
+
+
 # Description
 # - Returns the geometry of the parcel associated with a given address.
 # Accepts
 # - address: str
 # Returns
 # - parcel: str
 def get_parcel(address):
+    # Convert input to uppercase
+    check_addr = address.upper()
+    # Make a list of acceptable address strings - ["EAST 11TH STREET", "E 11TH STREET"], for example
+    acceptable = [check_addr]
+    # Check the input string for each of the cardinal directions
+    for dir, abbrev in cardinal_directions.items():
+        if dir in check_addr:
+            # Replace the first instance of the direction with its abbreviation
+            dir_normalized = check_addr.replace(dir, abbrev, 1)
+        # If this dir isn't in check_addr and dir_normalized hasn't been set:
+        elif "dir_normalized" not in locals():
```

Comment:
could you initialize dir_normalized to None outside of this for loop and then just check if it's none here like
```py
    dir_normalized = None
    for dir, abbrev in cardinal_directions.items():
        ....
        elif dir_normalized is None:
```

The reason I ask is that I would say `locals()` is an advanced concept for most people and the other way would be more accessible to other contributors.

That said, it's awesome you know about locals().  I'm sure there will be cases in the future where this knowledge comes in handy.